src/chart/sankey/SankeyView.js

```diff
@@ -61,38 +61,60 @@ function fadeInItem(item, opacityPath) {
     el.highlight && el.highlight();
 }
 
+function sum(array, cb, orient) {
+    var sum = 0;
+    var len = array.length;
+    var i = -1;
+    while (++i < len) {
+        var value = +cb.call(array, array[i], orient);
+        if (!isNaN(value)) {
+            sum += value;
+        }
+    }
```

Comment:
Please try this es6 for..in looping syntax:

var a =[ 3, 2, 1]
for(index in a){
    // do something with it.
    console.log(index);
}

Reply:
I'm afraid we can not use `for in` in array traverse, for one thing, it's limited by [ECharts coding-standard](https://echarts.apache.org/en/coding-standard.html#others)(and just ES3 is allowed), and for another, it's reasonable why we don't use as far as possible.

Reply:
Yes, we don't use ES6 features. Please use `zrUtil.forEach` instead.

Reply:
@Ovilia done